pixeltable/exec/cache_prefetch_node.py

```diff
@@ -253,12 +250,12 @@ def __fetch_url(self, url: str) -> tuple[Optional[Path], Optional[Exception]]:
                         }
                         self.boto_client = get_client(**config)
                 self.boto_client.download_file(parsed.netloc, parsed.path.lstrip('/'), str(tmp_path))
-            elif parsed.scheme == 'http' or parsed.scheme == 'https':
+            elif parsed.scheme in {'http', 'https'}:
```

Comment:
Is this actually required? My expectation is that this would be slightly less efficient. Is there any reason for this to be a set?

Reply:
I tried it as a tuple first and the linter corrected me.

I'm not an expert in Python performance optimizations, so I tend to trust the linter unless there's specific evidence it's misleading us.

Here's what ruff says.
https://docs.astral.sh/ruff/rules/repeated-equality-comparison/

Reply:
A well designed programming language will be well optimized and not use hash functions to test for membership of small sets.

For example, scala uses direct comparison on sets of cardinality <= 4 and only uses hashes if the cardinality is > 4.

I'm not an expert on Python performance and I don't know what Python does and I'll reiterate that I think we should default to trusting the experts unless we know otherwise.